lib/tesla/middleware/follow_redirects.ex

```diff
@@ -85,4 +86,14 @@ defmodule Tesla.Middleware.FollowRedirects do
     |> URI.merge(location)
     |> URI.to_string()
   end
+
+  # See https://github.com/teamon/tesla/issues/362
+  @filter_headers ["authorization"]
+  defp filter_headers(env, orig_url) do
```

Comment:
### cases

Several cases not handled

- header name in different case: e.g. both `authorization` `Authorization` should be handled. Maybe they're already lowercased before hitting here?
- same host but different port or scheme: e.g. `https://example.com` => `http://example.com:8080` - should drop the authorization

### userinfo in URL

Also.. is it possible that url in location (from redirect response) can have `userinfo`? If so, what should we do here?

1. request https://example.com with authorization
2. redirect to https://foo:bar@example.com/foo

In this case - should we drop original authorization *and* userinfo in the url?

### avoiding multiple URI.parse

Seems like it calls `URI.parse` somewhere - do you think we can refactor to avoid parsing multiple times?

We may make that as next refactoring PR - but just for the note...

Reply:
> header name in different case
headers are downcased by adapters

> same host but different port or scheme 
✅ 

> avoiding multiple URI.parse
✅ 

> userinfo in URL
I _think_ that if the location header includes the userinfo we should not drop it. I'd leave it for now.